pkg/volume/plugins.go

```diff
@@ -317,6 +318,23 @@ func NewSpecFromPersistentVolume(pv *api.PersistentVolume, readOnly bool) *Spec
 	}
 }
 
+// NewSpecFromVolumeWithNS creates an Spec from an api.Volume and a namespace
+func NewSpecFromVolumeWithNS(vs *api.Volume, nameSpace string) *Spec {
```

Comment:
This is really dirty: extending an existing object and then creating a separate constructor that initializes that field.